admin/javascript/src/tests/silverstripe-backend-test.js

```diff
@@ -163,5 +163,29 @@ describe('SilverStripeBackend', () => {
         expect(result).toEqual({ status: 'ok', message: 'happy' });
       });
     });
+
+    it('should replace url template parameters', () => {
+      const mock = getBackendMock({
+        text: () => Promise.resolve('{"status":"ok"'),
+        headers: new Headers({
+          'Content-Type': 'application/json',
+        }),
+      });
+      const endpoint = mock.createEndpointFetcher({
+        url: 'http://example.com/:type/:id/?foo=bar',
+        method: 'post',
+        urlTemplateParams: {
```

Comment:
Just thinking about future-proofing this a bit... I expect that we'll have other hooks to express what we want to do with the input. I wonder if this kind of approach might be more extensible?

    payloadSchema: {
      id: { urlReplacement: ':id', remove: true }
      type: { urlReplacement: ':id' }
      foo: { querystring: true }
    }

For example, in the example above I've said that `id` should be removed from the main payload but `type` should still be included, and that `foo` should be added to the querystring.